zipline/data/bundles/core.py

```diff
@@ -563,11 +563,16 @@ def should_clean(name):
                     (after is not None and dt > after)
                 )
 
-        else:
+        elif keep_last > 0:
             last_n_dts = set(all_runs[-keep_last:])
```

Comment:
we could use `toolz.take(keep_last, reversed(all_runs))` and then check >= 0 and cut the special case on `0`